### PR TITLE
use pk as default lookup_field for NautobotUIViewSet

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #3497 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by jathanism — APPROVED** [review]

File: changes/3607.removed [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed slug as the default lookup_field for NautobotUIViewSet making pk the default value.
```
Comment (glennmatthews) [inline changes/3607.removed:1]:
```suggestion
Removed `"slug"` as the default `lookup_field` for `NautobotUIViewSet`, making `"pk"` now the default value.
```

**Review by glennmatthews — CHANGES_REQUESTED** [review]
There's text in `nautobot/docs/plugins/development.md` about `lookup_field` that needs to be updated appropriately for this change.

File: nautobot/docs/plugins/development.md [hunk 2]
```diff
@@ -1037,19 +1037,19 @@ class YourAppModelUIViewSet(NautobotUIViewSet):
...
 
 ```python
 from nautobot.apps.views import NautobotUIViewset
 
 class YourAppModelUIViewSet(NautobotUIViewSet):
     ...
-    lookup_field = "pk"
+    lookup_field = "slug"
     ...
 ```
```
Comment (glennmatthews) [inline nautobot/docs/plugins/development.md:1050]:
````suggestion
```

+/- 2.0.0
    The default `lookup_field` for `NautobotUIViewSet` has been changed from `"slug"` to `"pk"`.
````

File: nautobot/docs/plugins/development.md [hunk 3]
```diff
@@ -1048,6 +1048,8 @@ class YourAppModelUIViewSet(NautobotUIViewSet):
...
 
++/- 2.0.0
+    The default `lookup_field` for `NautobotUIViewSet` has been changed from `"slug"` to `"pk"`.
 !!! note
```
Comment (glennmatthews) [inline nautobot/docs/plugins/development.md:1053]:
```suggestion
    The default `lookup_field` for `NautobotUIViewSet` has been changed from `"slug"` to `"pk"`.

!!! note
```

Comment (glennmatthews) [reply nautobot/docs/plugins/development.md:1053]: I think there's a bug in github's handling of suggested changes that include empty lines...

**Pull request merged**